models/avatars/avatar.go

```diff
@@ -109,39 +106,69 @@ func HashedAvatarLink(email string, size int) string {
 		}
 		return lowerEmail, nil
 	})
-	if size > 0 {
-		return setting.AppSubURL + "/avatar/" + url.PathEscape(sum) + "?size=" + strconv.Itoa(size)
-	}
-	return setting.AppSubURL + "/avatar/" + url.PathEscape(sum)
+	return emailHash
 }
 
-// MakeFinalAvatarURL constructs the final avatar URL string
-func MakeFinalAvatarURL(u *url.URL, size int) string {
-	vals := u.Query()
-	vals.Set("d", "identicon")
-	if size != DefaultAvatarSize {
-		vals.Set("s", strconv.Itoa(size))
+// GenerateUserAvatarFastLink returns a fast link to the user's avatar via the local explore page.
+func GenerateUserAvatarFastLink(userName string, size int) string {
+	if size < 0 {
+		size = 0
```

Comment:
Why is the size here lower bounded to `0`, and in line 126 below to `DefaultAvatarSize`? I think these two should be set to the same value.

Reply:
`0` and `-1` have same meaning in related logic, all will be default avatar size.

I prefer to use 0 here because the generated URL `/user/avatar/user-name/0` is more beautiful than `/user/avatar/user-name/-1`

Reply:
Now I removed the useless constant `DefaultAvatarSize`. Only `0` .